### ISSUE TITLE
"UNKNOWNs: munin graph is unknown." in alert mail

### ISSUE THREAD

**Issue by cweiske** [munin-monitoring/munin#888]
I'm using Munin 2.0.33-1~bpo8+1 (Debian 8) with CGI graphing (`html_strategy cgi`).
Every night at 23:14 I get a mail with one warning:

    Munin processing time
    "UNKNOWNs: munin graph is unknown."

Is there a way to turn this off?

In 2012 someone opened a ticket about the same topic, without an answer: http://munin-monitoring.org/ticket/1169

**Comment by sumpfralle** [comment]
I am using 2.0.33-1 (Debian 9) - thus there is probably no difference with regard to the code.
I also configured `html_strategy cgi`.
These error messages do not appear on my system.
Thus we should try to find the difference between your setup and mine.

Could it be that there are some old configuration settings around on your system?

For me:
```shell
$ grep -RA 3 munin_stats /etc/munin
/etc/munin/plugin-conf.d/munin-node:[munin_stats]
/etc/munin/plugin-conf.d/munin-node-user munin
/etc/munin/plugin-conf.d/munin-node-group munin
/etc/munin/plugin-conf.d/munin-node-
```

Do you have other ideas, what could be different?

**Comment by cweiske** [comment]
```
$ grep -RA 3 munin_stats /etc/munin
/etc/munin/plugin-conf.d/munin-node:[munin_stats]
/etc/munin/plugin-conf.d/munin-node-user munin
/etc/munin/plugin-conf.d/munin-node-group munin
/etc/munin/plugin-conf.d/munin-node-
--
/etc/munin/plugin-conf.d/munin-node.dpkg-dist:[munin_stats]
/etc/munin/plugin-conf.d/munin-node.dpkg-dist-user munin
/etc/munin/plugin-conf.d/munin-node.dpkg-dist-group munin
/etc/munin/plugin-conf.d/munin-node.dpkg-dist-
```

**Comment by cweiske** [comment]
@sumpfralle: Munin shows "nan" for "munin graph" and "munin html" in the web interface. Do you have that, too?

![munin_stats-day](https://user-images.githubusercontent.com/59036/36684963-38504880-1b22-11e8-8880-66a44c98c6be.png)

**Comment by sumpfralle** [comment]
Regarding `nan`: you are using the CGI (on-demand) generation of HTML files and diagrams - thus there is no tracking of the processing time.

Regarding your error message (`UNKNOWNs: munin graph is unknown.`): do you have any other idea? Are there cron-Jobs running around that time on your system?

**Comment by cweiske** [comment]
Now I know where it is coming from:
- Every 5 minutes, `/usr/bin/munin-cron` is executed.
- Every day 23:14, `/usr/share/munin/munin-limits --force` is executed.

That mail is being sent every day at 23:14, so munin-limits must be the source.

**Comment by cweiske** [comment]
Now I see that I should have probably listed shown full mail:

```
Subject: Munin notification ahso
Date: Mon, 26 Feb 2018 23:14:01 +0100 (CET)

ahso2 :: ahso2 :: Connections through firewall

ahso2 :: ahso2 :: Disk usage in percent

ahso2 :: ahso2 :: Inode usage in percent

ahso2 :: ahso2 :: beancounter for CT76996: privvmpages

ahso2 :: ahso2 :: Postfix statistics

ahso2 :: ahso2 :: Munin processing time
	UNKNOWNs: munin graph is unknown.

ahso2 :: ahso2 :: venet0 errors

ahso2 :: ahso2 :: File table usage
```

**Comment by cweiske** [comment]
```
$ munin-limits --debug
...
2018/02/27 06:20:25 [DEBUG] processing service: munin_stats
2018/02/27 06:20:25 [DEBUG] state_file: /var/lib/munin/state-ahso2-ahso2.storable
2018/02/27 06:20:25 [DEBUG] processing critical: update ->  : 285
2018/02/27 06:20:25 [DEBUG] processing warning: update ->  : 240
2018/02/27 06:20:25 [DEBUG] processing unknown_limit: update -> 3

2018/02/27 06:20:25 [DEBUG] processing field: ahso2::ahso2::munin_stats::update
2018/02/27 06:20:25 [DEBUG] field: $VAR1 = {
  '#%#name' => 'update',
  'critical' => '285',
  'draw' => 'AREASTACK',
  'graph_data_size' => 'normal',
  'label' => 'munin update',
  'update_rate' => '300',
  'warning' => '240'
};
2018/02/27 06:20:25 [DEBUG] value: ahso2::ahso2::munin_stats::update: 14.68 (crit: :285) (warn: :240)
2018/02/27 06:20:25 [DEBUG] processing critical: graph ->  : 285
2018/02/27 06:20:25 [DEBUG] processing warning: graph ->  : 240
2018/02/27 06:20:25 [DEBUG] processing unknown_limit: graph -> 3

2018/02/27 06:20:25 [DEBUG] processing field: ahso2::ahso2::munin_stats::graph
2018/02/27 06:20:25 [DEBUG] field: $VAR1 = {
  '#%#name' => 'graph',
  'critical' => '285',
  'draw' => 'AREASTACK',
  'graph_data_size' => 'normal',
  'label' => 'munin graph',
  'update_rate' => '300',
  'warning' => '240'
};
2018/02/27 06:20:25 [DEBUG] value: ahso2::ahso2::munin_stats::graph: unknown (crit: :285) (warn: :240)
2018/02/27 06:20:25 [DEBUG] processing unknown_limit: html -> 3
2018/02/27 06:20:25 [DEBUG] processing unknown_limit: limits -> 3

2018/02/27 06:20:25 [DEBUG] generating service message: ahso2::ahso2::munin_stats
2018/02/27 06:20:25 [DEBUG] Contact list for ahso2::ahso2::munin_stats: cweiske

2018/02/27 06:20:25 [DEBUG] processing service: if_err_venet0
...
```

**Comment by sumpfralle** [comment]
Thank you for your additional input: now I found the problem.
The `warning/critical` config lines for "graph" may not be enabled, if the graphs are generated via CGI.

This was already fixed in master and is now fixed in `stable-2.0`, as well.
